weave/weave_server.py

```diff
@@ -379,16 +380,47 @@ def frontend(path):
 @blueprint.route("/", defaults={"path": None})
 @blueprint.route("/<path:path>")
 def root_frontend(path):
+    if request.args.get("unsetBetaVersion") is not None:
+        resp = redirect_without_qs_param("unsetBetaVersion")
+        resp.set_cookie("betaVersion", "", max_age=0)
+        return resp
+
+    new_beta_version = request.args.get("betaVersion")
+    if new_beta_version is not None:
+        resp = redirect_without_qs_param("betaVersion")
+        resp.set_cookie("betaVersion", new_beta_version)
+        return resp
+
     # To support server cases where we're mounted under an existing path, i.e.
     # /wandb/weave, then index.html will load something like /wandb/weave/.../env.js
     if path is not None:
         path = path.split("/")[-1]
     if path == "env.js":
         js = f"window.WEAVE_CONFIG = {json.dumps(frontend_env())}"
         return Response(js, mimetype="application/javascript")
+
+    beta_version = request.cookies.get("betaVersion")
+    if beta_version is not None:
+        beta_version = beta_version[:9]
+        content = requests.get(
+            f"https://cdn.wandb.ai/weave/{beta_version}/index.html",
+            stream=True,
+        ).content
+        return Response(content, mimetype="text/html")
+
     return send_from_directory(blueprint.static_folder, "index.html")
 
 
+def redirect_without_qs_param(param: str):
```

Comment:
**nitpick:** What's qs? might be better to name explicitly, sth like redirect without query param?

Reply:
Ah my b. I just assumed that qs means querystring to most people, but that's just because I'm a dirty web dev. `query` is probably better, changed it!